design/design.md

```diff
@@ -1,46 +1,31 @@
 ## Design Decisions:
-For the design of our React To-Do List, we mainly took inspiration from the previous lab's static HTML/CSS To-Do List. However, this time, we utilized more of Professor Milburn's basic design principles.
+The largest design decision made was regarding how to store our data (the structure of the firestore schema). We know that we want a user to be able to grab all of their to-do list items. We noticed that in future labs, we have to have the ability to support individual users having their own lists separate from others (instead of a singular global list shared amongst all users). Furthermore, every user must be able to have multiple lists. Thus, we preemptively prepared for these cases. Our structure is as follows: we have a Users collection and for each user document, we have an multiple subcollections, where each one represents a list. For now, a user document just consists of a name property and an array that stores the string names of the user's lists. The default global user has a name of "user" with id "User". Within the lists subcollections, there are item documents that have the properties id, text, completed, priority, and created. Thus, the path to the global list of item documents would look something like "Users/User/Items" if one of the lists was called "Items".
```

Comment:
You'll find that in Lab 5, however, you also need to be able to share lists with other users. You may find it easier to have a structure of Lists/Items where each List has a userId.